src/components/case-card.js

```diff
@@ -132,20 +142,35 @@ const StakeLocked = styled.div`
   text-align: right;
 `;
 
-const PeriodCard = ({ period, deadline, hiddenVotes }) => {
-  const phases = ["Evidence Submission", "Commit Deadline", "Voting Deadline", "Appeal Deadline", "Execute Deadline"];
-  const periodText = Number(period) === 2 && hiddenVotes ? "Reveal Deadline" : phases[period];
+const phases = ["Evidence Submission", "Commit Deadline", "Voting Deadline", "Appeal Deadline", "Execute Deadline"];
+
+const PeriodCard = ({ period, deadline, hiddenVotes, isVoteCommitted }) => {
+  let periodText = phases[period];
+  const showCommited = Number(period) === 1 && isVoteCommitted;
+
+  if (Number(period) === 2 && hiddenVotes) {
+    periodText = "Reveal Deadline";
+  } else if (showCommited) {
+    periodText = "Committed ✅";
+  }
```

Comment:
I think I would do it like this: (pseudocode, check if it does what you want it to do and pass it through the linter)
```suggestion
  const showCommited = useMemo(() => Number(period) === 1 && isVoteCommited, [period, isVoteCommited]);
  const periodText = useMemo(() => {
    const showRevealDeadline = Number(period) === 2 && hiddenVotes;
    
    if (showRevealDeadline) {
      return "Reveal Deadline";
    } else if (showCommited) {
      return "Committed ✅";
    } else {
      return phases[period];
    }
  }, [showCommited, period, hiddenVotes, phases]);
```